honeybee/shade.py

```diff
@@ -0,0 +1,284 @@
+            factor: A number representing how much the object should be scaled.
+            origin: A ladybug_geometry Point3D representing the origin from which
+                to scale. If None, it will be scaled from the World origin (0, 0, 0).
+        """
+        self._geometry = self.geometry.scale(factor, origin)
+
+    def check_planar(self, tolerance, raise_exception=True):
+        """Check whether all of the Shade's vertices lie within the same plane.
+
+        Args:
+            tolerance: The minimum distance between a given vertex and a the
+                object's's plane at which the vertex is said to lie in the plane.
+            raise_exception: Boolean to note whether an ValueError should be
+                raised if a vertex does not lie within the object's plane.
+        """
+        try:
+            return self.geometry.validate_planarity(tolerance, raise_exception)
+        except ValueError as e:
+            raise ValueError('Shade "{}" is not planar.\n{}'.format(
+                self.name_original, e))
+
+    def check_self_intersecting(self, raise_exception=True):
+        """Check whether the edges of the Shade intersect one another (like a bowtwie).
+
+        Args:
+            raise_exception: If True, a ValueError will be raised if the object
+                intersects with itself. Default: True.
+        """
+        if self.geometry.is_self_intersecting:
+            if raise_exception:
+                raise ValueError('Shade "{}" has self-intersecting edges.'.format(
+                    self.name_original))
+            return False
+        return True
+
+    def check_non_zero(self, tolerance=0.0001, raise_exception=True):
+        """Check whether the area of the Shade is above a certain "zero" tolerance.
+
+        Args:
+            tolerance: The minimum acceptable area of the object. Default is 0.0001,
+                which is equal to 1 cm2 when model units are meters. This is just
+                above the smalest size that OpenStudio will accept.
+            raise_exception: If True, a ValueError will be raised if the object
+                area is below the tolerance. Default: True.
+        """
+        if self.area < tolerance:
+            if raise_exception:
+                raise ValueError(
+                    'Shade "{}" geometry is too small. Area must be at least {}. '
+                    'Got {}.'.format(self.name_original, tolerance, self.area))
+            return False
+        return True
+
+    @property
+    def properties(self):
+        """Shade properties, including Radiance, Energy and other properties."""
+        return self._properties
+
+    @property
+    def to(self):
+        """Shade writer object.
+
+        Use this method to access Writer class to write the shade in different formats.
+
+        Usage:
+            shade.to.idf(shade) -> idf string.
+            shade.to.radiance(shade) -> Radiance string.
+        """
+        return writer
```

Comment:
This should probably be set to `NotImplementedError()` until we implement the writers.